modules/user/user.repository.js

```diff
@@ -0,0 +1,43 @@
+const express = require("express");
+const fsPromises = require("fs").promises;
+const filePath = "./database/users.json";
+
+class UserRepository {
+    async readFileUsers() {
+        let data = await fsPromises.readFile(filePath);
+        const usersArr = await JSON.parse(data);
+        return usersArr;
+    }
+
+    findUserIndexById(usersArr, id) {
```

Comment:
your repository function don't need to be depend on `userArr`, so you need to get `useArr` inside of `findUserIndexById` function so this should look something like 
`const userArr = await this.readFileUsers()`

- [ ] all you repository functions shouldn't depend on usersArr